code/datums/components/thermite.dm

```diff
@@ -0,0 +1,72 @@
+/datum/component/thermite
+	var/static/list/blacklist
+	var/static/list/resistlist
+	var/amount
+
+/datum/component/thermite/Initialize(_amount)
+	. = ..()
+	blacklist = typecacheof(/turf/closed/wall/mineral/diamond)
+	resistlist = typecacheof(/turf/closed/wall/r_wall)
+
+	if(blacklist[parent.type])
+		qdel(src)
+		return
+	if(resistlist[parent.type])
+		_amount*=0.25
+
+	amount = _amount*10
+
+	if(!istype(parent, /turf))
+		qdel(src)
+		return
+
+	var/turf/master = parent
+	master.cut_overlays()
```

Comment:
If the wall has its own overlays, will this remove them permanently?

Reply:
Yes

Reply:
should probably remove explicitly the thermite overlay